app/controllers/api/groups_controller.rb

```diff
@@ -142,11 +142,34 @@ def update_marks
 
     def set_mark_by_criteria(criteria, mark_to_change)
```

Comment:
I don't like my use of branching if statements here, but I can't see a simpler way.